src/components/Microsoft.DotNet.UpgradeAssistant.MSBuild/MSBuildProject.File.cs

```diff
@@ -158,6 +158,41 @@ public void RenameFile(string filePath)
         public void AddItem(string name, string path)
             => ProjectRoot.AddItem(name, path);
 
+        public bool RemoveItem(ProjectItemType? itemType = null, string? includePath = null, string? excludePath = null, string? elementName = null)
+        {
+            var items = ProjectRoot.Items.Where(item =>
+            {
+                if (itemType != null && item.ItemType != itemType.Name)
+                {
+                    return false;
+                }
+
+                if (includePath != null && item.Include != includePath)
+                {
+                    return false;
+                }
+
+                if (excludePath != null && item.Exclude != excludePath)
+                {
+                    return false;
+                }
+
+                if (elementName != null && item.ElementName != elementName)
+                {
+                    return false;
+                }
+
+                return true;
+            });
```

Comment:
Can we use `??` check instead of explicit `null` checks? Also why is there the need to return something on every failure? a return for `if items.any()` should suffice, no?

Reply:
Noted. I'll make this change.

Reply:
Feels like this would be far more readable as a fluent set of `.Where()` clauses instead of one large one in addition to conveying the filters are additive